core/src/test/java/io/neow3j/protocol/rx/JsonRpc2_0RxTest.java

```diff
@@ -0,0 +1,296 @@
+        Disposable disposable = observable.subscribe(
+            result -> {
+                results.add(result);
+                transactionLatch.countDown();
+            },
+            throwable -> fail(throwable.getMessage()),
+            () -> completedLatch.countDown());
+
+        transactionLatch.await(1, TimeUnit.SECONDS);
+        assertThat(results, equalTo(neoGetBlocks));
+
+        disposable.dispose();
+
+        completedLatch.await(1, TimeUnit.SECONDS);
+        assertTrue(disposable.isDisposed());
+    }
+
+    @Test
+    public void testReplayBlocksDescendingObservable() throws Exception {
+
+        List<NeoGetBlock> neoGetBlocks = Arrays
+            .asList(createBlock(2), createBlock(1), createBlock(0));
+
+        OngoingStubbing<NeoGetBlock> stubbing =
+            when(neow3jService.send(any(Request.class), eq(NeoGetBlock.class)));
+        for (NeoGetBlock neoGetBlock : neoGetBlocks) {
+            stubbing = stubbing.thenReturn(neoGetBlock);
+        }
+
+        Observable<NeoGetBlock> observable = neow3j.replayBlocksObservable(
+            new BlockParameterIndex(BigInteger.ZERO),
+            new BlockParameterIndex(BigInteger.valueOf(2)),
+            false,
+            false);
+
+        CountDownLatch transactionLatch = new CountDownLatch(neoGetBlocks.size());
+        CountDownLatch completedLatch = new CountDownLatch(1);
+
+        List<NeoGetBlock> results = new ArrayList<>(neoGetBlocks.size());
+        Disposable disposable = observable.subscribe(
+            result -> {
+                results.add(result);
+                transactionLatch.countDown();
+            },
+            throwable -> fail(throwable.getMessage()),
+            () -> completedLatch.countDown());
+
+        transactionLatch.await(1, TimeUnit.SECONDS);
+        assertThat(results, equalTo(neoGetBlocks));
+
+        disposable.dispose();
+
+        completedLatch.await(1, TimeUnit.SECONDS);
+        assertTrue(disposable.isDisposed());
+    }
+
+    @Test
+    public void testCatchUpToLatestAndSubscribeToNewBlockObservable() throws Exception {
+
+        List<NeoGetBlock> expected = Arrays.asList(
+            createBlock(0),
+            createBlock(1),
+            createBlock(2),
+            createBlock(3),
+            createBlock(4),
+            createBlock(5),
+            createBlock(6)
+        );
+
+        List<NeoGetBlock> neoGetBlocks = Arrays.asList(
+            // past blocks:
+            expected.get(0),
+            expected.get(1),
+            expected.get(2),
+            expected.get(3),
+            // later blocks:
+            expected.get(4),
+            expected.get(5),
+            expected.get(6)
+        );
+
+        OngoingStubbing<NeoGetBlock> stubbingNeoGetBlock =
+            when(neow3jService.send(any(Request.class), eq(NeoGetBlock.class)));
+
+        for (int i = 0; i < 7; i++) {
+            stubbingNeoGetBlock = stubbingNeoGetBlock.thenReturn(neoGetBlocks.get(i));
```

Comment:
Why is `neoGetBlocks` used here instead of `expected`?

Reply:
Addressed.